src/argilla_sdk/datasets/_export.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+import json
+import logging
+import os
+from pathlib import Path
+import warnings
+from typing import Optional, Union, TYPE_CHECKING, Tuple
+from uuid import uuid4
+
+from argilla_sdk._models import DatasetModel
+from argilla_sdk.client import Argilla
+from argilla_sdk.settings import Settings
+from argilla_sdk.workspaces._resource import Workspace
+
+
+if TYPE_CHECKING:
+    from argilla_sdk import Dataset
+    from argilla_sdk.records import DatasetRecords
+
+
+class DiskImportExportMixin(ABC):
+    """A mixin for exporting and importing datasets to and from disk."""
+
+    _model: DatasetModel
+    _default_settings_path = "settings.json"
+    _default_records_path = "records.json"
+    _default_dataset_path = "dataset.json"
+
+    def to_disk(self, path: str) -> str:
+        """Exports the dataset to disk in the given path. The dataset is exported as a directory containing the dataset model, settings and records as json files.
+
+        Args:
+            path (str): The path to export the dataset to. Must be an empty directory.
+        """
+
+        dataset_path, settings_path, records_path = self._define_child_paths(path=path)
+
+        # Deal with path creation and use
+        if os.path.exists(path):
+            if not os.path.isdir(path):
+                raise ValueError(f"Path {path} is not a directory")
+            if os.listdir(path):
+                raise ValueError(f"Directory {path} is not empty")
+        else:
+            os.makedirs(name=path, exist_ok=True)
+
+        # Export the dataset model, settings and records
+        with open(dataset_path, "w") as f:
+            json.dump(self._model.model_dump(), f)
```

Comment:
Since the private method `_persist_dataset_model` does the same, we could use it.
```suggestion
        self._persist_dataset_model(dataset_path)
```